crates/biome_js_analyze/src/semantic_analyzers/nursery/no_misleading_character_class.rs

```diff
@@ -171,11 +170,23 @@ impl Rule for NoMisleadingCharacterClass {
                 }
             }
             AnyRegexExpression::JsCallExpression(expr) => {
-                let callee = match expr.callee().ok()? {
-                    AnyJsExpression::JsIdentifierExpression(callee) => callee,
+                let is_regexp = match expr.callee().ok()? {
```

Comment:
The logic of `is_regexp` is completely same, so it woulld be better to make it the local function.